lib/std/wordwrap.nim

```diff
@@ -0,0 +1,67 @@
+import unicode
+
+proc wordWrap*(s: string, maxLineWidth = 80,
+               splitLongWords = true,
+               newLine = "\n"): string  =
+  ## This function breaks all words that reach over `maxLineWidth`
+  ## measured in number of runes. When `splitLongWords` is `true`
+  ## words that are longer than `maxLineWidth` are splitted. Multiple
+  ## spaces and newlines are converted to a single space.  All
+  ## whitespace is treated equally.  Non-breaking whitespace is
+  ## ignored.
+
+  var currentWordLength: int = 0
+  var currentWord: string = newStringOfCap(32)
+  var currentLineLength: int = 0
+  var currentWordLengthAtLineEnd: int = -1
+  var longWordMode = false
+
+  template handleWhitespace(): untyped =
+    if currentWord.len > 0:
+
+      if currentLineLength + 1 + currentWordLength > maxLineWidth:
+        result.add newLine
```

Comment:
```suggestion
        if len(result) > 0:
          result.add newLine
```
otherwise you are inserting an empty line, if the length of the first word on the first line is >= maxLineWidth.